### PR TITLE
Updates to Config utils to compare slices and adding tags to networking config structs

### PR DESCRIPTION
**Pull request by SenRamakri**
This PR contains changes from @edhemphill to implement support in Config Utils to detect delta in  golang "slices" and corresponding tests. In addition I have changes to add tags to networking config structs and additions to configUtils to split the CallChanges into Diff and CallChanges. All the tests are passing with these changes and should be ok to merge.

### CODE REVIEW

File: configUtils.go [hunk 1]
```diff
@@ -246,27 +246,101 @@ func (a *ConfigAnalyzer) CallChanges(current interface{}, future interface{}) (i
...
-				// different values - add to list
-				group, ok := field.Tag.Lookup(a.configTag)
-				if ok {
-					c, ok2 := allchanges[group]
-					if !ok2 {
-						c = new(changes)
-						allchanges[group] = c
-						c.configgroup = group
+			// For slices we treat the slice as a single value,
+			// unless the slice is a slice of struct or ptr to struct
```
Comment (cmonr) [inline configUtils.go:244]: It bothers me that we can now have an object that is _either_ a slice, or contain a set of slices.

Would it be possible to instead always return a set of slices, even if that set only has one member?

Comment (cmonr) [reply configUtils.go:244]: Imo, we should avoid actively determining the structure of an object at runtime, as I feel that could lead to issues down the line.

**Review by cmonr — COMMENTED** [review]
Looks alright to me, but left a comment that I would like reviewed/addressed first.

File: configUtils.go (hunk 1 above)
Comment (cmonr) [reply configUtils.go:244]: Talked w/ @SenRamakri, and he explained this.

**Comment by cmonr** [comment]
Making a note. @SenRamakri found a bug and is in the middle of fixing.

**Comment by SenRamakri** [comment]
> Making a note. @SenRamakri found a bug and is in the middle of fixing.

@cmor - I have added my new changes. Please review and merge.

**Review by cmonr — APPROVED** [review]

**Pull request merged**